### DISCUSSION TITLE
A simple Python script to make repeated DMM measurements . . .

### DISCUSSION THREAD

**Discussion by dbbotkin** [ni/nimi-python#1805] [Ideas]
import nidmm
import time

PXI4071 = nidmm.Session("PXI-4071")
PXI4071.configure_measurement_digits(nidmm.Function.DC_VOLTS, 10, 7.5)
for i in range(1, 5001):
  t = (str(PXI4071.get_dev_temp()))
  print(str(i) +', ' + t + ', ' + str(PXI4071.read()))
  time.sleep(3)

I use this simple routine to monitor characteristics of metrology references e.g. Fluke 732a. There are also times when I use Python to coordinate multiple instruments taking measurements.

**Comment by marcoskirsch** [comment]
> # {this doesn't improve measurement; it should.}

Just set `configure_measurement_digits` as you do, and NI-DMM will set all the lower level settings needed in order to accomplish that, including aperture time, ADC calibration, etc.

Consider using `with` statement or explicitly calling `close()` so that you don't keep the instrument session open.

**Comment by dbbotkin** [comment]
Right. I’ll try the close() . I assume that is to close the session at the end of the script.
